compiler/rustc_hir_analysis/src/check/intrinsic.rs

```diff
@@ -121,10 +121,10 @@ pub fn intrinsic_operation_unsafety(tcx: TyCtxt<'_>, intrinsic_id: DefId) -> hir
     if has_safe_attr != is_in_list {
         tcx.sess.struct_span_err(
             tcx.def_span(intrinsic_id),
-            DiagnosticMessage::Str(format!(
+            DiagnosticMessage::from(format!(
                     "intrinsic safety mismatch between list of intrinsics within the compiler and core library intrinsics for intrinsic `{}`",
-                    tcx.item_name(intrinsic_id)
-        ))).emit();
+                    tcx.item_name(intrinsic_id))
+        )).emit();
```

Comment:
This formatting (and the moved `)`) doesn't sit right with me. Shouldn't it be more like
```rust
tcx.sess.struct_span_err(
    tcx.def_span(intrinsic_id),
    DiagnosticMessage::from(format!(
        "intrinsic safety mismatch between list of intrinsics within the compiler and core library intrinsics for intrinsic `{}`",
        tcx.item_name(intrinsic_id)
    ))
).emit();
```

Reply:
:+1: